libevmasm/RuleList.h

```diff
@@ -629,6 +630,12 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		feasibilityFunction
 	});
 
+	rules.push_back({
```

Comment:
As @ekpyron [mentioned](https://github.com/ethereum/solidity/issues/9119#issuecomment-1028060749) in the original issue, `A, B, C...` is for constants, and `X, Y, Z, ...` for arbitrary values - so will this rule then cover the `and(shl(A, X), shl(A, Y))` as well?

Reply:
Yes it will. constants are part of arbitrary values :)